_posts/2020-07-08-pull-requests-require-context.md

```diff
@@ -0,0 +1,148 @@
+---
+layout: epic
+title: "Pull Requests Need Context"
+date: "2020-07-08"
+author: [steve-hicks]
+categories: [tools, github, team, community, engineering]
+---
+
+Code review is an engineering process that has benefited greatly from a move toward asynchronous communication.
+Long ago, engineering teams would sit around in a room together with code on a projector to review changes. 😱 For
+many teams this would mean batching code reviews or even skipping them altogether. 😱😱
+
+Today, most engineering teams use incredible tools like GitHub or GitLab to review changes through Pull Requests
+(PRs). The greatest advantage of PRs is that the review can happen when it's convenient for the reviewer:
+asynchronously.
+
+<!-- more -->
+
+One important feature _missing_ from asynchronous communication is the ability to course-correct when context is
+misunderstood. When you're in a synchronous conversation with someone, it doesn't take much time for them to let
+you know you've forgotten to include context. It doesn't take much time for you to add the missing context to keep
+the conversation moving forward.
+
+It takes a lot longer to identify missing context when communicating asynchronously. Worse, lack of context when
+_reviewing code_ asynchronously has a reverb effect. I create my PR when it's convenient for me, you ask a
+clarifying question when it's convenient for you, I respond when it's convenient for me, etc. Suddenly my PR has
+been open for three days and we haven't yet made it to a common understanding of why I've made these changes.
+
+It's extremely important to include all available context when drafting a PR. It saves incredible amounts of time
+by cutting out slow round-trip conversations to clarify.
+
+I'm personally proud of and impressed by the job we do at Artsy in including context in our PRs. We start early, by
+giving our engineers
+[some reading about how we work with PRs during their onboarding](https://github.com/artsy/README/blob/master/playbooks/engineer-workflow.md#pull-requests).
+
+But beyond that our engineers lead by example. This article presents a handful of examples from Artsy repositories
+demonstrating how you can add context to your PRs to avoid unnecessary clarifying conversation.
+
+## Explain Your Reasoning
+
+You've been thinking a lot about the problem you're solving - probably significantly more than your reviewers.
+You'll save everyone time by describing the problem and sharing how you're thinking about it.
+
+### Define the problem and solution
+
+Why does this PR exist? Explain the problem it solves and describe your solution, as
+[Sarah](https://github.com/sweir27) does [in this PR](https://github.com/artsy/force/pull/3095). For bonus points,
+include alternative approaches you considered.
```

Comment:
I personally love a basic PR template with a structure like this to encourage authors to explain the context!